ant/base/ant.py

```diff
@@ -177,13 +183,13 @@ def _worker(self):
                     # Channel event
                     elif (
                         message._id == Message.ID.RESPONSE_CHANNEL
-                        and message.data[1] == 0x01
+                        and message._data[1] == 0x01
                     ):
                         _logger.debug("Got channel event, %r", message)
                         self._events.put(
                             (
                                 "event",
-                                (message._data[0], message._data[2], message._data[3:]),
+                                (message._data[0], message._data[1], message._data[2:]),
```

Comment:
Shouldn't this be `(message._data[0], message._data[2], message._data[3:]),` as it where? aka channel (0), event (2) and rest data?

Reply:
This line changed in your last commit, and it raised an exception on receiving a channel event, because message._data array had only 3 bytes : (channel, 1,event_code) (ant protocol §9.5.6.1). And the rest of your code, especially wait_for_event in filter.py, expects the event tuple to be exactly this. The filter.py code is a bit hard to read, but this tuple is translated to (channel, event,data), with event always = 1 (channel event, not response), and data[0] corresponding to event_code. The code works even if event parameters are sent, which my device doesn't do.